src/plugins/intel_npu/src/plugin/npuw/llm_infer_request.cpp

```diff
@@ -100,6 +105,20 @@ void copy_columns_by_row_chunks(ov::SoPtr<ov::ITensor> src, ov::SoPtr<ov::ITenso
         std::copy_n(src_p + src_offset, chunk_byte_size, dst_p + dst_offset);
     }
 }
+
+std::optional<ov::Output<const ov::Node>> find_port_by_name(const std::vector<ov::Output<const ov::Node>>& ports,
+                                                            const std::string& name) {
+    auto it = std::find_if(ports.begin(), ports.end(), [&](const auto& port) {
+        return port.get_names().count(name) != 0;
+    });
+    if (it == ports.end()) {
+        return std::nullopt;
+    }
+    return std::make_optional(*it);
+}
```

Comment:
y do you need optional here, while it is is just a little syntax sugar of the `find_if` so the result could be an interator?